src/app/src/components/FacilityListItemsFilterSearch.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState } from 'react';
+import { func, string } from 'prop-types';
+
+import FormControl from '@material-ui/core/FormControl';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import TextField from '@material-ui/core/TextField';
+import Close from '@material-ui/icons/Close';
+import Search from '@material-ui/icons/Search';
+
+import { ENTER_KEYCODE, ESCAPE_KEYCODE } from '../util/constants';
+
+export default function FacilityListItemsTableSearch({
+    currentValue,
+    onSearch,
+}) {
+    const [searchText, setSearchText] = useState(currentValue);
+    const isCurrent = searchText === currentValue;
+
+    // Update internal state if the prop is updated
+    useEffect(() => setSearchText(currentValue), [currentValue]);
+
+    const handleSearch = () => {
+        if (isCurrent) {
+            return;
+        }
+
+        onSearch(searchText);
+    };
+
+    const handleTextChange = (e) => {
+        setSearchText(e.target.value);
```

Comment:
Forgot to mention this but there's a `getValueFromEvent` utility method for this. It's debatable whether this is more readable but this could potentially just be

```
const handleTextChange = flow(getValueFromEvent, setSearchText);
```

Reply:
I find the current version to be more readable, will keep that. Thanks for surfacing that though!